src/main/java/minijava/Compiler.java

```diff
@@ -0,0 +1,309 @@
+package minijava;
+
+import static minijava.Cli.dumpGraphsIfNeeded;
+
+import com.google.common.io.Files;
+import com.google.common.util.concurrent.Runnables;
+import firm.ClassType;
+import firm.Entity;
+import firm.Graph;
+import firm.MethodType;
+import firm.Program;
+import firm.Type;
+import firm.Util;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import minijava.ir.assembler.allocator.OnTheFlyRegAllocator;
+import minijava.ir.assembler.block.AssemblerFile;
+import minijava.ir.emit.IREmitter;
+import minijava.ir.optimize.AlgebraicSimplifier;
+import minijava.ir.optimize.CommonSubexpressionElimination;
+import minijava.ir.optimize.ConstantControlFlowOptimizer;
+import minijava.ir.optimize.ConstantFolder;
+import minijava.ir.optimize.CriticalEdgeDetector;
+import minijava.ir.optimize.ExpressionNormalizer;
+import minijava.ir.optimize.FloatInTransformation;
+import minijava.ir.optimize.Inliner;
+import minijava.ir.optimize.JmpBlockRemover;
+import minijava.ir.optimize.LoadStoreOptimizer;
+import minijava.ir.optimize.Optimizer;
+import minijava.ir.optimize.OptimizerFramework;
+import minijava.ir.optimize.PhiOptimizer;
+import minijava.ir.optimize.ProgramMetrics;
+import minijava.ir.optimize.UnreachableCodeRemover;
+import minijava.lexer.Lexer;
+import minijava.parser.Parser;
+import minijava.semantic.SemanticAnalyzer;
+import minijava.semantic.SemanticLinter;
+import minijava.token.Token;
+import minijava.util.PrettyPrinter;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.lambda.tuple.Tuple2;
+
+public class Compiler {
+
+  public static Lexer lex(InputStream in) {
```

Comment:
I think I know what you mean, but I disagree. `Iterator` really strongly suggests laziness. Also `map`, `filter` et al. are finally accepted nowadays and it's the same thing there.